service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -287,8 +290,20 @@ public Workspace updateWorkspace(
       @Nullable String userFacingId,
       @Nullable String name,
       @Nullable String description,
+      @Nullable ApiTpsPolicyInputs addAttributes,
+      @Nullable ApiTpsPolicyInputs removeAttributes,
+      @Nullable ApiTpsUpdateMode updateMode,
       AuthenticatedUserRequest userRequest) {
     if (workspaceDao.updateWorkspace(workspaceUuid, userFacingId, name, description)) {
+      if (addAttributes != null || removeAttributes != null) {
```

Comment:
This ends up being two transactions, so it is possible for the update part to succeed, the TPS part to fail. The response is failure, but the workspace part of the update still happened.

I think the choice is:
1. Make a separate endpoint and you do not need to use flights
2. Use the combined endpoint, but make this a two step flight.

Reply:
Yeah, that gets to the question I had of should this be a flight or not. I think the best approach is to separate it out, then it really will mirror the tps update endpoint...

Reply:
sgtm